### DISCUSSION TITLE
Question: How to implement/use middleware feature like in nuxt.js ?

### DISCUSSION THREAD

**Discussion by sriks82** [hannoeru/vite-plugin-pages#374] [Q&A]
Is it possible to create src/middleware/guest.js, src/middleware/auth.js, src/middleware.role.js... and define the middleware in pages ?

For example, is a page exists like: src/pages/student/somepage.vue then this page should be accessible only by an authenticated user and the role of the user should be `student`.

**Comment by olcchi** [comment]
I had the same idea, but the answer I got so far is no , But there are corresponding methods to achieve the same function, such as `extendRoute`

**Comment by marekvospel** [comment]
vite-plugin-pages is not an ssr framework, so it cannot act exactly like nuxt.js (serving as server side api routes)

You can however define [`beforeEach` guards](https://router.vuejs.org/api/interfaces/router.html#beforeeach) after creating the router. See an example:
```ts
router.beforeEach((to, from, next) => {
    if (!to.fullPath.startsWith('/admin')) return next()

    const me = getMe()
    if (me.hasPermission(Permission.ADMIN)) return next()

    next('/')
  })
```
